code/go/0chain.net/smartcontract/partitions/blobber_list.go

```diff
@@ -0,0 +1,159 @@
+package partitions
+
+import (
+	"0chain.net/chaincore/chain/state"
+	"0chain.net/core/common"
+	"0chain.net/core/datastore"
+	"0chain.net/core/util"
+	"encoding/json"
+	"fmt"
+)
+
+//------------------------------------------------------------------------------
+
+type BlobberNode struct {
+	ID  string `json:"id"`
+	Url string `json:"url"`
+}
+
+func (bn *BlobberNode) Encode() []byte {
+	var b, err = json.Marshal(bn)
+	if err != nil {
+		panic(err)
+	}
+	return b
+}
+
+func (bn *BlobberNode) Decode(b []byte) error {
+	return json.Unmarshal(b, bn)
+}
+
+func (bn *BlobberNode) Data() string {
+	return bn.Url
+}
+
+func (bn *BlobberNode) Name() string {
+	return bn.ID
+}
+
+//------------------------------------------------------------------------------
+
+type blobberItemList struct {
+	Key     datastore.Key `json:"-"`
+	Items   []BlobberNode `json:"items"`
+	Changed bool          `json:"-"`
+}
+
+func (il *blobberItemList) Encode() []byte {
+	var b, err = json.Marshal(il)
+	if err != nil {
+		panic(err)
+	}
+	return b
+}
+
+func (il *blobberItemList) Decode(b []byte) error {
+	return json.Unmarshal(b, il)
+}
+
+func (il *blobberItemList) save(balances state.StateContextI) error {
+	_, err := balances.InsertTrieNode(il.Key, il)
+	return err
+}
+
+func (il *blobberItemList) get(key datastore.Key, balances state.StateContextI) error {
+	val, err := balances.GetTrieNode(key)
+	if err != nil {
+		if err != util.ErrValueNotPresent {
+			return err
+		}
+		il = &blobberItemList{
+			Key: key,
+		}
+	}
+	if err := il.Decode(val.Encode()); err != nil {
+		return fmt.Errorf("%w: %s", common.ErrDecoding, err)
+	}
+	il.Key = key
+	return nil
+}
+
+func (il *blobberItemList) add(it PartitionItem) {
+	il.Items = append(il.Items, BlobberNode{
+		ID:  it.Name(),
+		Url: it.Data(),
+	})
+	il.Changed = true
+}
+
+func (il *blobberItemList) update(it PartitionItem) error {
+	for i := range il.itemRange(0, il.length()) {
+		if il.Items[i].Name() == it.Name() {
+			var newItem BlobberNode
+			err := newItem.Decode(it.Encode())
```

Comment:
We can do a type assertion and then do a copy to avoid unnecessary data encoding/decoding, such as:

bn, ok :=  (*it.BlobberNode)
if !ok{
  return err 
}

newItem :=  *bn
// add the newItem to list.